### PR TITLE
Z-Level Falling

### PR DESCRIPTION
**Pull request by Geevies**
* Falling down a Z-Level without rolling now has a chance to dislocate the limb you land on.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/multiz/movement.dm [hunk 1]
```diff
@@ -476,6 +476,32 @@
 		visible_message("<span class='warning'>\The [src] falls and lands directly on their legs!</span>",
 			"<span class='danger'>You land on your feet, and the impact brings you to your knees.</span>")
 
+		if(prob(20))
+			var/obj/item/organ/external/l_foot = get_organ("l_foot")
+			var/obj/item/organ/external/r_foot = get_organ("r_foot")
+
+			if(prob(50) && l_foot)
+				visible_message("<span class='warning'>There's a sickening popping noise as [src]'s left ankle bends unnaturally!</span>",
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]: use the span("notice", "

File: code/modules/multiz/movement.dm [hunk 2]
```diff
@@ -476,6 +476,32 @@
...
+				"<span class='danger'>Grueling pain shoots through your mind as your left ankle dislocates under the fall!</span>")
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]: here too

File: code/modules/multiz/movement.dm [hunk 3]
```diff
@@ -476,6 +476,32 @@
...
+				l_foot.dislocate(TRUE)
+			else if(r_foot)
+				visible_message("<span class='warning'>There's a sickening popping noise as [src]'s right ankle bends unnaturally!</span>",
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]: and here

File: code/modules/multiz/movement.dm [hunk 4]
```diff
@@ -476,6 +476,32 @@
...
+				"<span class='danger'>Grueling pain shoots through your mind as your right ankle dislocates under the fall!</span>")
+				r_foot.dislocate(TRUE)
+		else if(prob(15))
+			var/obj/item/organ/external/l_leg = get_organ("l_leg")
+			var/obj/item/organ/external/r_leg = get_organ("r_leg")
+
+			if(prob(50) && l_leg)
+				visible_message("<span class='warning'>There's a sickening popping noise as [src]'s left knee caves in!</span>",
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]:
```suggestion
				visible_message(span("warning", "There's a sickening popping noise as [src]'s left knee caves in!")),
```

File: code/modules/multiz/movement.dm [hunk 5]
```diff
@@ -476,6 +476,32 @@
...
+				"<span class='danger'>Grueling pain shoots through your mind as your left knee dislocates under the fall!</span>")
+				l_leg.dislocate(TRUE)
+			else if(r_leg)
+				visible_message("<span class='warning'>There's a sickening popping noise as [src]'s right knee caves in!</span>",
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]: here

File: code/modules/multiz/movement.dm [hunk 6]
```diff
@@ -476,6 +476,32 @@
...
+				"<span class='danger'>Grueling pain shoots through your mind as your right knee dislocates under the fall!</span>")
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]: here

File: code/modules/multiz/movement.dm [hunk 7]
```diff
@@ -496,12 +522,43 @@
 
 		visible_message("<span class='warning'>\The [src] falls and lands arms first!</span>",
```
Comment (alsoandanswer) [inline code/modules/multiz/movement.dm]: yeah i think you know the drill

File: code/modules/multiz/movement.dm [hunk 8]
```diff
@@ -473,8 +473,34 @@
...
+			var/obj/item/organ/external/r_foot = get_organ("r_foot")
+
+			if(prob(50) && l_foot)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s left ankle bends unnaturally!"),
+				span("danger", "Grueling pain shoots through your mind as your left ankle dislocates under the fall!"))
+				l_foot.dislocate(TRUE)
+			else if(r_foot)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s right ankle bends unnaturally!"),
+				span("danger", "Grueling pain shoots through your mind as your right ankle dislocates under the fall!"))
+				r_foot.dislocate(TRUE)
```
Comment (NonQueueingMatt) [inline code/modules/multiz/movement.dm]: you can condense this I'm pretty sure, it's duplicate code

File: code/modules/multiz/movement.dm [hunk 9]
```diff
@@ -473,8 +473,34 @@
...
+			var/obj/item/organ/external/r_leg = get_organ("r_leg")
+
+			if(prob(50) && l_leg)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s left knee caves in!"),
+				span("danger", "Grueling pain shoots through your mind as your left knee dislocates under the fall!"))
+				l_leg.dislocate(TRUE)
+			else if(r_leg)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s right knee caves in!"),
+				span("danger", "Grueling pain shoots through your mind as your right knee dislocates under the fall!"))
+				l_leg.dislocate(TRUE)
```
Comment (NonQueueingMatt) [inline code/modules/multiz/movement.dm]: as above

File: code/modules/multiz/movement.dm [hunk 10]
```diff
@@ -494,13 +520,44 @@
...
+			var/obj/item/organ/external/r_hand = get_organ("r_hand")
+
+			if(prob(50) && l_hand)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s left wrist bends unnaturally!"),
+				span("danger", "Grueling pain shoots through your mind as your left wrist dislocates under the fall!"))
+				l_hand.dislocate(TRUE)
+			else if(r_hand)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s right wrist bends unnaturally!"),
+				span("danger", "Grueling pain shoots through your mind as your right wrist dislocates under the fall!"))
+				r_hand.dislocate(TRUE)
```
Comment (NonQueueingMatt) [inline code/modules/multiz/movement.dm]: as above

File: code/modules/multiz/movement.dm [hunk 11]
```diff
@@ -494,13 +520,44 @@
...
+			var/obj/item/organ/external/r_arm = get_organ("r_arm")
+
+			if(prob(50) && l_arm)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s left elbow caves in!"),
+				span("danger", "Grueling pain shoots through your mind as your left elbow dislocates under the fall!"))
+				l_arm.dislocate(TRUE)
+			else if(r_arm)
+				visible_message(span("warning", "There's a sickening popping noise as [src]'s right elbow caves in!"),
+				span("danger", "Grueling pain shoots through your mind as your right elbow dislocates under the fall!"))
+				r_arm.dislocate(TRUE)
```
Comment (NonQueueingMatt) [inline code/modules/multiz/movement.dm]: as above

**Pull request merged**